packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,600 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'icons.dart';
+import 'text_selection.dart';
+
+export 'package:flutter/services.dart' show TextInputType, TextInputAction, TextCapitalization;
+
+const BorderSide _kDefaultRoundedBorderSide = BorderSide(
+  color: CupertinoColors.lightBackgroundGray,
+  style: BorderStyle.solid,
+  width: 0.0,
+);
+const Border _kDefaultRoundedBorder = Border(
+  top: _kDefaultRoundedBorderSide,
+  bottom: _kDefaultRoundedBorderSide,
+  left: _kDefaultRoundedBorderSide,
+  right: _kDefaultRoundedBorderSide,
+);
+const BoxDecoration _kDefaultRoundedBorderDecoration = BoxDecoration(
+  border: _kDefaultRoundedBorder,
+  borderRadius: BorderRadius.all(Radius.circular(4.0)),
+);
+
+const TextStyle _kDefaultTextStyle = TextStyle(
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.38,
+  color: CupertinoColors.black,
+  decoration: TextDecoration.none,
+);
+
+const Color _kSelectionHighlightColor = Color(0x667FAACF);
+const Color _kInactiveTextColor = Color(0xFFC2C2C2);
+const Color _kDisabledBackground = Color(0xFFFAFAFA);
+
+/// Visibility of text field overlays based on the state of the current text entry.
+enum OverlayVisibilityMode {
+  /// Overlay will never appear regardless of the text entry state.
+  never,
+  /// Overlay will only appear when the current text entry is not empty.
+  ///
+  /// This includes pre-filled text that the user did not type in manually. But
+  /// does not include text in placeholders.
+  editing,
+  /// Overlay will only appear when the current text entry is empty.
+  ///
+  /// This also includes not having pre-filled text that the user did not type
+  /// in manually. Texts in placeholders are ignored.
+  notEditing,
+  /// Always show the overlay regardless of the text entry state.
+  always,
+}
+
+/// An iOS-style text field.
+///
+/// A text field lets the user enter text, either with hardware keyboard or with
+/// an onscreen keyboard.
+///
+/// This widget corresponds to both a [UITextField] and an editable [UITextView]
+/// on iOS.
+///
+/// The text field calls the [onChanged] callback whenever the user changes the
+/// text in the field. If the user indicates that they are done typing in the
+/// field (e.g., by pressing a button on the soft keyboard), the text field
+/// calls the [onSubmitted] callback.
+///
+/// To control the text that is displayed in the text field, use the
```

Comment:
I think a couple of code snippets here would probably be much more clear than a description of what to do.  You can show a code snippet for setting an initial value, and a code snippet for selecting text.

Reply:
Good point. Adding